linera-execution/src/runtime.rs

```diff
@@ -944,28 +1031,27 @@ impl ServiceRuntime for ServiceSyncRuntime {
         queried_id: UserApplicationId,
         argument: Vec<u8>,
     ) -> Result<Vec<u8>, ExecutionError> {
-        let (query_context, code) = {
-            let mut this = self.as_inner();
+        let (query_context, service) = {
+            let mut this = self.inner();
 
             // Load the application.
-            let (code, description) = this.load_service(queried_id)?;
+            let (service, parameters) = this.load_service_instance(queried_id)?;
             // Make the call to user code.
             let query_context = crate::QueryContext {
                 chain_id: this.chain_id,
             };
             this.push_application(ApplicationStatus {
                 id: queried_id,
-                parameters: description.parameters,
```

Comment:
If we're going to keep the full description in the map, I would leave the field access here.

Reply:
I refactored to create a `LoadedApplication` helper type that contains just the parameters from the description. I think this makes it a bit clearer and avoids cloning the entire description when instancing an application.